CMakeLists.txt

```diff
@@ -64,7 +69,33 @@ deprecated_option(OLD NETCDF_ACTIVE         NEW NGEN_WITH_NETCDF)
 deprecated_option(OLD BMI_C_LIB_ACTIVE      NEW NGEN_WITH_BMI_C)
 deprecated_option(OLD BMI_FORTRAN_ACTIVE    NEW NGEN_WITH_BMI_FORTRAN)
 deprecated_option(OLD QUIET                 NEW NGEN_QUIET)
-deprecated_option(OLD PACKAGE_TESTS         NEW NGEN_WITH_TESTS)
+
+# -- Removed
+# deprecated_option(OLD PACKAGE_TESTS         NEW NGEN_WITH_TESTS)
+
+# Options to automate building of extern-distributed models
+option(NGEN_WITH_EXTERN_SLOTH                            "Build with extern-distributed SLoTH"            ON)
+cmake_dependent_option(NGEN_WITH_EXTERN_TOPMODEL         "Build with extern-distributed TOPMODEL"         ON "NGEN_WITH_BMI_C"       OFF)
+cmake_dependent_option(NGEN_WITH_EXTERN_CFE              "Build with extern-distributed CFE"              ON "NGEN_WITH_BMI_C"       OFF)
+cmake_dependent_option(NGEN_WITH_EXTERN_PET              "Build with extern-distributed PET"              ON "NGEN_WITH_BMI_C"       OFF)
+cmake_dependent_option(NGEN_WITH_EXTERN_NOAH_OWP_MODULAR "Build with extern-distributed Noah-OWP-Modular" ON "NGEN_WITH_BMI_FORTRAN" OFF)
+
+# Allow enabling of all extern models dependent on NGen options given.
+option(NGEN_WITH_EXTERN_ALL OFF)
+if(NGEN_WITH_EXTERN_ALL)
+    set(NGEN_WITH_EXTERN_SLOTH ON)
+
+    if(NGEN_WITH_BMI_C)
```

Comment:
I think if someone has asked for "ALL", it really should mean "all". Thus, if we're missing language support for some external model that we're going to build, I think that should be an error. We may also want to just hard-require Python support as well in this case - i.e. that any compiled Python modules that would be necessary have been compiled and found.